### PR TITLE
Add Error Prone

### PR DESCRIPTION
**Pull request by NightlyNexus**
This is going to fail until https://github.com/google/error-prone/commit/bee0ce7f3b8989ee7883c0db152cc0491444a1a9 is released, but is this even something you'd consider having in the project?

### CODE REVIEW

**Comment by mattprecious** [comment]
❤️ error prone

**Comment by NightlyNexus** [comment]
Bah, I don't see why Error Prone is still failing at the case fall-through. I'll have to check again later/soon.

File: build.gradle [hunk 1]
```diff
@@ -14,4 +24,13 @@ allprojects {
   version = VERSION_NAME
 
   apply plugin: 'maven'
+  apply plugin: 'net.ltgt.errorprone'
+
+  tasks.withType(JavaCompile) {
+    options.compilerArgs += ['-Xlint:all',
+                             '-Xlint:-fallthrough',
```
Comment (NightlyNexus) [inline build.gradle]: I'm so dumb, it was just javac xlint failing.

File: telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java [hunk 2]
```diff
@@ -59,7 +59,7 @@
  */
 public class TelescopeLayout extends FrameLayout {
   private static final String TAG = "Telescope";
-  static final SimpleDateFormat SCREENSHOT_FILE_FORMAT =
```
Comment (mattprecious) [inline telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java]: Uhh, why doesn't error prone like this?

Comment (NightlyNexus) [inline telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java]: http://errorprone.info/bugpattern/SimpleDateFormatConstant
It's not a constant, as the name implies.

Comment (JakeWharton) [inline telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java]: Hmm weird. We just suppressed this internally with a comment that it's accesses were thread-confined. If only the `@MainThread` annotation allowed field targets... http://b.android.com/208133

Comment (NightlyNexus) [inline telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java]: nice. i'll suppress it.

**Pull request closed without merging**